GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -141,6 +141,31 @@ string ComposeOutput()
                 () => executable.RunCommandAsync(arguments, input, createWindow));
         }
 
+        /// <summary>
+        /// Launches a process for the executable and returns <c>true</c> if its exit code is zero. Fix #6593.
+        /// </summary>
+        /// <remarks>
+        /// This method uses <see cref="RunCommand"/> to execute multiple commands in batch, used in accordance with
+        /// <see cref="ArgumentBuilderExtensions.BuildBatchArguments(ArgumentBuilder, IEnumerable{string}, int, int)"/>
+        /// to work around Windows command line length 32767 character limitation
+        /// <see href="https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-createprocessa"/>.
+        /// </remarks>
+        /// <param name="executable">The executable from which to launch a process.</param>
+        /// <param name="batchArguments">The array of batch arguments to pass to the executable</param>
+        /// <param name="input">Bytes to be written to the process's standard input stream, or <c>null</c> if no input is required.</param>
+        /// <param name="createWindow">A flag indicating whether a console window should be created and bound to the process.</param>
+        /// <returns><c>true</c> if the process's exit code was zero, otherwise <c>false</c>.</returns>
+        [MustUseReturnValue("Callers should verify that " + nameof(RunBatchCommand) + " returned true")]
+        public static bool RunBatchCommand(
+            this IExecutable executable,
+            ArgumentString[] batchArguments,
+            byte[] input = null,
+            bool createWindow = false)
+        {
+            // TODO: Process somehow always return non-zero exit code.
+            return batchArguments.Aggregate(true, (result, arguments) => result && executable.RunCommand(arguments, input, createWindow));
```

Comment:
`result && executable.RunCommand(...)`: In C/C++, the function would never be called due to the lazy evaluation of logical expressions and because `result` is initialized with `true`. Though C# seems to work differently: In the debugger, I can see that `RunCommand` _is_ called. @gitextensions/git-extensions-source, can we rely on it?

Reply:
I'm gonna change this to a for loop, both for clarity and for batch progress callback needed in #7237.

Reply:
I've switched function execution before boolean var, so in any language (C/C++), it would look logical now. The for loop change would be in another PR.